### PR TITLE
Changes in E2E tests

### PR DESCRIPTION
**Pull request by BOHEUS**
Changes introduced in #6539 were overwritten in this PR because:
1. the test checks if table is visible in Welcome page, not in Companies page (_because even though there's_ `goto('/objects/companies')`, _tests are run from the fresh setup so user is not logged in and there is redirect back to `/welcome` page due to fallback mechanism which activates every time, when unauthorized user tries to access a page behind login page (can be easily reproduced when going to_ `/objects/people`_before logging in), which is clearly documented in linked PR_) so the whole point of the test is basically misunderstood
2. playwright.config.ts is basically generated template with added imported config from dotenv and setting up baseURL, where there should be some more settings, like e.g. screenshots on failure, fixed viewport (resolution of browser to ensure consistency between tests and environments) and such
3. .env contains only base URL which can be set in playwright.config.ts (but that's up to taste) whereas there should be at least some more data e.g. user's login and password

Added e2e login test (#6637)
Reverted changes from #6621 as they caused silent failure of app (#6639)
Added Github Workflow CI (for now it'll always fail as it doesn't have proper data for logging) basic driver for running commands in OS shell, basic .env.example with variables, necessary options to configuration (#6641)
Added tests mentioned in #6467

### CODE REVIEW

File: packages/twenty-e2e-testing/tests/companies.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      workerInfo.project.name +
+      `${date.toISOString()}.png`,
+  });
+});
+
+test.describe('Basic check', () => {
+  test('Checking if table in Companies is visible', async ({ page }) => {
+    await expect(page.getByTestId('tooltip').nth(0)).toHaveText('Companies');
+    await expect(page.getByTestId('tooltip').nth(0)).toBeVisible();
+    expect(page.url()).toContain('/companies');
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/companies.spec.ts:30]: For now, by default user is redirected to Companies tab after logging in, once changes from #6629 are merged, it'll be fixed

File: packages/twenty-e2e-testing/.env.example [hunk 2]
```diff
@@ -1,2 +1,20 @@
 # Note that provide always without trailing forward slash to have expected behaviour
 FRONTEND_BASE_URL="http://localhost:3001"
+DEFAULT_LOGIN=tim@apple.dev
+NEW_WORKSPACE_LOGIN=test@apple.dev
+DEMO_DEFAULT_LOGIN=noah@demo.dev
+DEFAULT_PASSWORD=Applecar2025
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/.env.example:6]: Default passwords aren't a security risk as they're available in documentation

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      await page.getByRole('button', { name: 'Continue' }).click();
+      await page.getByText('Continue without sync').click();
+      await page.getByRole('button', { name: 'Finish' }).click();
+      await expect(page.locator('table')).toBeVisible({ timeout: 1000 });
+    }
+  });
+
+  test('Syncing all workspaces', async () => {
+    await sh('npx nx run twenty-server:command workspace:sync-metadata -f');
+    await sh('npx nx run twenty-server:command workspace:sync-metadata -f');
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts:49]: In #6467 it was mentioned to run sync twice

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    await page.getByPlaceholder('Password').fill('Applecar2025');
+    await page.getByRole('button', { name: 'Sign in' }).click();
+    await page.getByRole('link', { name: 'Opportunities' }).click();
+    await expect(page.locator('tbody > tr')).toHaveCount(4);
+    expect(page.url()).not.toContain('/welcome');
+  });
+
+  test('Creating new workspace', async ({ page, browserName }) => {
+    // this test must use only 1 browser, otherwise it will lead to success and fail (1 workspace is created instead of x workspaces)
+    if (browserName == 'firefox') {
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts:27]: There is comment explaining why it's Firefox-only (ideally should be 1 browser/worker only, not specific browser only)

File: packages/twenty-e2e-testing/tests/companies.spec.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  await page.screenshot({
+    path:
+      `./packages/twenty-e2e-testing/results/screenshots/${browserName}/` +
+      workerInfo.project.name +
+      `${date.toISOString()}.png`,
+  });
+});
+
+test.describe('Basic check', () => {
+  test('Checking if table in Companies is visible', async ({ page }) => {
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/companies.spec.ts:27]: Will be fixed later in the future

File: packages/twenty-e2e-testing/playwright.config.ts [hunk 6]
```diff
@@ -1,43 +1,73 @@
...
+  retries: process.env.CI ? 2 : 0,
+  workers: process.env.CI ? 1 : undefined, // undefined = amount of projects
+  timeout: 30 * 1000, // timeout can be changed
   use: {
-    /* Base URL to use in actions like `await page.goto('/')`. */
-    baseURL: process.env.FRONTEND_BASE_URL ?? 'http://localhost:3001',
+    baseURL:
+      process.env.CI ??
+      process.env.FRONTEND_BASE_URL ??
+      'http://localhost:3001',
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/playwright.config.ts:24]: This should be `process.env.CI ? process.env.DEMO_DEFAULT_BASE_URL : process.env.FRONTEND_BASE_URL ?? 'http://localhost:3001'`

File: packages/twenty-e2e-testing/tests/companies.spec.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { config } from 'dotenv';
+import path = require('path');
+config({ path: path.resolve(__dirname, '..', '.env') });
+
+const date = new Date();
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/');
+  await page.getByRole('button', { name: 'Continue With Email' }).click();
+  await page.getByPlaceholder('Email').fill(process.env.DEFAULT_LOGIN);
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/companies.spec.ts:11]: This must stay like this so anyone can run tests after editing .env

File: packages/twenty-e2e-testing/README.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
 # Twenty e2e Testing
 
-## Install
+## Prerequisition
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/README.md:3]: Fixed

File: packages/twenty-e2e-testing/config/customreporter.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  TestResult,
+  FullResult,
+} from '@playwright/test/reporter';
+
+class CustomReporter implements Reporter {
+  constructor(options: { customOption?: string } = {}) {
+    console.log(
+      `my-awesome-reporter setup with customOption set to ${options.customOption}`,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/config/customreporter.ts:15]: WIP as whether custom reporter should be used when there are different reporters provided by the framework

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+      await page.goto('/');
+      await page.getByRole('button', { name: 'Continue With Email' }).click();
+      await page.getByPlaceholder('Email').fill('test@apple.dev'); // email must be changed each time test is run
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts:30]: This either needs to be replaced with `process.env.NEW_WORKSPACE_LOGIN` or stay like this

**Comment by lucasbordeau** [comment]
@BOHEUS Could you please handle #6639 in a separate PR ?

**Comment by lucasbordeau** [comment]
We should remove the CI workflow part for now, please understand that we don't need and we can't afford it for now, because it would involve a lot of CI minutes for something we only need to run manually while deploying a new version.

File: packages/twenty-e2e-testing/playwright.config.ts [hunk 11]
```diff
@@ -1,43 +1,72 @@
...
+    baseURL: process.env.CI
+      ? process.env.CI_DEFAULT_BASE_URL
+      : (process.env.FRONTEND_BASE_URL ?? 'http://localhost:3001'),
+    trace: 'retain-on-failure', // trace takes EVERYTHING from page source, records every single step, should be used only when normal debugging won't work
+    screenshot: 'on', // either 'on' here or in different method in modules, if 'on' all screenshots are overwritten each time the test is run
+    headless: true, // instead of changing it to false, run 'yarn test:e2e:debug' or 'yarn test:e2e:ui'
+    testIdAttribute: 'data-testid', // taken from Twenty source
+    viewport: { width: 1920, height: 1080 }, // most laptops use this resolution
+    launchOptions: {
+      slowMo: 500, // time in milliseconds between each step, better to use it than explicitly define timeout in tests
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/playwright.config.ts:30]: Nice

File: packages/twenty-e2e-testing/playwright.config.ts [hunk 12]
```diff
@@ -1,43 +1,72 @@
...
     {
       name: 'firefox',
       use: { ...devices['Desktop Firefox'] },
     },
 
-    {
-      name: 'webkit',
-      use: { ...devices['Desktop Safari'] },
-    },
+    //{
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/playwright.config.ts:47]: Should we remove dead code ?

File: packages/twenty-e2e-testing/tests/companies.spec.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import { test, expect } from '@playwright/test';
+import { config } from 'dotenv';
+import path = require('path');
+config({ path: path.resolve(__dirname, '..', '.env') });
+
+const date = new Date();
+
+test.beforeEach(async ({ page }) => {
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/companies.spec.ts:8]: Should we extract this in a login module as you proposed or do you plan to do it in another PR ?

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    await page.getByRole('button', { name: 'Continue' }).click();
+    await page.getByPlaceholder('Password').fill('Applecar2025');
+    await page.getByRole('button', { name: 'Sign in' }).click();
+    await page.getByRole('link', { name: 'Opportunities' }).click();
+    await expect(page.locator('tbody > tr')).toHaveCount(4);
+    expect(page.url()).not.toContain('/welcome');
+  });
+
+  test('Creating new workspace', async ({ page, browserName }) => {
+    // this test must use only 1 browser, otherwise it will lead to success and fail (1 workspace is created instead of x workspaces)
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/workspaces.spec.ts:26]: Shouldn't we instead play the whole test on each brother while resetting the DB before each ?

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/workspaces.spec.ts:1]: Very nice basis for the future !

File: packages/twenty-e2e-testing/playwright.config.ts (hunk 12 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/playwright.config.ts:47]: That's not a dead code, I commented it out due to problems on my site, I can't install webkit driver for Playwright due to dependencies conflict with some Linux libraries, but anyway, it should stay.

File: packages/twenty-e2e-testing/tests/companies.spec.ts (hunk 13 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/companies.spec.ts:8]: I'll do it another PR

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts:1]: Thanks!

**Comment by BOHEUS** [comment]
Oh, didn't mean that

**Comment by BOHEUS** [comment]
@lucasbordeau I brought back changes from #6621, I'll look into #6639 and check if it's still actual, also, I removed workflow and saved it to .yml.bak in case tests will be run in the future

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts (hunk 14 above)
Comment (lucasbordeau) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts:26]: Ok let's use only chromium as we said earlier then ?

**Comment by lucasbordeau** [comment]
Hi @BOHEUS, quick follow-up : 
- You can ignore greptile comments
- Could you please open the next PR on another branch of your repo, because it's make it harder for us to push changes when we merge main for example.
- I'm testing this in local and will merge right after

**Comment by lucasbordeau** [comment]
My bad, seems like this optimizeDeps line has been removed in a previous PR, so we can remove it !

File: packages/twenty-front/vite.config.ts [hunk 15]
```diff
@@ -108,5 +108,8 @@ export default defineConfig(({ command, mode }) => {
         localsConvention: 'camelCaseOnly',
       },
     },
+    optimizeDeps: {
```
Comment (lucasbordeau) [inline packages/twenty-front/vite.config.ts:111]: Let's remove this

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+      await page.getByPlaceholder('Password').fill('Applecar2025');
+      await page.getByRole('button', { name: 'Sign in' }).click();
+      await page.getByRole('link', { name: 'Companies' }).click();
+      expect(page.url()).toContain('/companies');
+      await expect(page.locator('table')).toBeVisible();
+    }
+  });
+
+  test('Seeding database', async ({ page, browserName }) => {
+    if (browserName === 'firefox') {
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/workspaces.spec.ts:68]: Shouldn't we use chromium ?

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  test('Syncing all workspaces', async () => {
+    await sh('npx nx run twenty-server:command workspace:sync-metadata -f');
+    await sh('npx nx run twenty-server:command workspace:sync-metadata -f');
+  });
+
+  test('Resetting database', async ({ page, browserName }) => {
+    if (browserName === 'firefox') {
+      await sh('yarn nx database:reset twenty-server'); // if this command fails for any reason, database must be restarted manually using the same command because database is in unstable state
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/workspaces.spec.ts:54]: Could we extract this in a beforeAll ? Because this command will take time and right now it can run in parallel.

Also it could probably cause bugs if the server is up and running because metadata are computed at runtime. We should tear down the server and restart it after reset and seed are completed, that way we're 100% sure that everything is the same as what we do in local when testing manually.

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    path: `./packages/twenty-e2e-testing/results/screenshots/${browserName}/${date.toISOString()}.png`,
+  });
+});
+
+test.describe('', () => {
+  test('Testing logging', async ({ page }) => {
+    await page.goto('/');
+    await page.getByRole('button', { name: 'Continue With Email' }).click();
+    await page.getByPlaceholder('Email').fill('tim@apple.dev');
+    await page.getByRole('button', { name: 'Continue' }).click();
```
Comment (lucasbordeau) [inline packages/twenty-e2e-testing/tests/workspaces.spec.ts:17]: The test is failing here for me, seems like it cannot get the Continue button.

**Comment by BOHEUS** [comment]
@lucasbordeau I'm closing this PR as it became way too messier than I expected, should've created PR from different branch than main, I'm sorry for this, I'll create another PR but this time properly and with fixes mentioned here as well as other issues.

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts (hunk 16 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts]: It was done just to ensure that only 1 browser will be running tests as I didn't implement it in playwright.config.ts, on which browser should it be run that's up to taste. I understand the question as Chrome is the most popular browser, I'll change it to Chromium (because it's better to check against the engine used in other browsers essentially rather than specific browser)

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts (hunk 18 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts]: That's weird, is it because the button is not active as the Email field wasn't triggered by the click? Please use `yarn test:e2e:ui workspaces` (or `yarn playwright test --debug workspaces`) to see where it fails

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts (hunk 17 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts]: You're right about possible corruption, that's why I did browser check, I'll fix this once I write down the correct flow of these tests.

File: packages/twenty-e2e-testing/tests/workspaces.spec.ts (hunk 18 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/workspaces.spec.ts]: Okay, that's because of optimizeDeps line, since there's blank page, test can't find button and can't log in.
